File: LeetCode/Climbing_stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        vector<int> result(n+2);
        result[0] = 1;
        result[1] = 1;
        result[2] = 2;
        for(int iter = 3;iter<=n;iter++){
            result[iter] = result[iter-1]+result[iter-2];
        }
        return result[n];
    }
};

File: LeetCode/climbing_stairs.java
class Solution {
    public int climbStairs(int n) {
        int[] steps = new int[n+1];
        if(n>2){
            steps[0] = 0;
            steps[1] = 1;
            steps[2] = 2;
            for(int iter = 3; iter <= n ;iter++){
                steps[iter] = steps[iter-1] + steps[iter-2];
            }
            
            return steps[n];
        }
        return n;
    }
}